src/app-layer-ssl.c

```diff
@@ -355,121 +353,6 @@ void SSLVersionToString(uint16_t version, char *buffer)
     }
 }
 
-static void TlsDecodeHSCertificateErrSetEvent(SSLState *ssl_state, uint32_t err)
-{
-    switch (err) {
-        case ERR_DER_UNKNOWN_ELEMENT:
-            SSLSetEvent(ssl_state,
-                        TLS_DECODER_EVENT_CERTIFICATE_UNKNOWN_ELEMENT);
-            break;
-        case ERR_DER_ELEMENT_SIZE_TOO_BIG:
-        case ERR_DER_INVALID_SIZE:
-        case ERR_DER_RECURSION_LIMIT:
-            SSLSetEvent(ssl_state,
-                        TLS_DECODER_EVENT_CERTIFICATE_INVALID_LENGTH);
```

Comment:
in my failing test this event is missing. Are we setting events based on the new decoder?

Reply:
At the moment, no. The decoder can raise errors (See list at https://docs.rs/x509-parser/0.6.5/x509_parser/error/enum.X509Error.html, and https://docs.rs/der-parser/3.0.4/der_parser/error/enum.BerError.html for the possible structure decoding errors), but at the moment they are all transformed into a `TLS_DECODER_EVENT_INVALID_CERTIFICATE` event.

It's not wrong, but maybe lacks some precision in the error. Are there signatures or detection rules that really requires this ? If yes, I can try to find a way to send more events back to C

Reply:
Looking at the precision provided by the Rust parsers, I would actually like to request exposing all of them. I think it would be great if we can match on those and log those in the anomaly logger.